glib/src/param_spec.rs

```diff
@@ -1370,6 +1370,77 @@ impl ParamSpecVariant {
     }
 }
 
+pub trait Parametrize {
+    type Spec;
```

Comment:
In my proposal I had multiple traits like this so that you can map the different cases: no default/range, only default, default and range.

And then there are strange cases like valuearray that take another pspec for the element type.

Reply:
I've seen you proposal, but I still wanted to try reusing the already existing Spec::new constructor. Probably I'll end up implementing things as in your proposal anyway...

Reply:
Writing this out for feedback...
Once we have a builder for each `ParamSpec`, this macro is going to call the builder with the respective parameters.
API example:
`#[prop(get, set, builder(min = 2, max = 4)]'

Why add `builder(...)`? Because we don't know beforehand what are the possible parameters (eg: GStreamer has custom ParamSpecs), and there shouldn't be a collision with the other macro's attributes.